### ISSUE TITLE
Cannot add application to Teleport

### ISSUE THREAD

**Issue by frodes-bre** [gravitational/teleport#21804]
Expected behavior:
To connect application from Teleport dashboard and be able to access it

Current behavior:
Have setup Teleport following this guide, https://goteleport.com/docs/try-out-teleport/linux-server/, and are now able to logon to Teleport dashboard and ssh to vm. Am now trying to add an application from the Teleport dashboard, but when running teleport start --config"/home/user/.config/app_config.yaml" I get this error:
2023-02-14T14:39:59Z ERRO [PROC:1]    App failed to establish connection to cluster: Post "https://xxxxxx:443/v1/webapi/host/credentials": dial tcp xxx.xxx.xxx.xx:443: connect: connection timed out. pid:4900.1 service/connect.go:119

Bug details:
- Teleport version: Teleport v12.0.2 git:api/v12.0.2-0-g64e23540e9 go1.19.5
- Recreation steps: Have tried to add app both on a seperate server and on the same server that is running the auth and proxy service
- Debug logs

Any suggestions?

**Comment by zmb3** [comment]
Can you share `app_config.yaml`? (Please redact any secrets first)

**Comment by frodes-bre** [comment]
Hi,
Did a reinstall, but still get the same errors.

Here is the app_config.yaml:
"version: v3
teleport:
  nodename: teleport-vm
  data_dir: /home/teleportuser/.config
  join_params:
    token_name: *******************************
    method: token
  proxy_server: *******.com:443
  log:
    output: stderr
    severity: INFO
    format:
      output: text
  ca_pin: ""
  diag_addr: ""
auth_service:
  enabled: "no"
ssh_service:
  enabled: "no"
proxy_service:
  enabled: "no"
  https_keypairs: []
  https_keypairs_reload_interval: 0s
  acme: {}
app_service:
  enabled: "yes"
  debug_app: false
  apps:
  - name: nrtest2
    uri: http://localhost:1880/
    public_addr: "nrtest2.*******.com"
    insecure_skip_verify: false"
    
Did also add the app when following the inital guide for setting up Teleport, but there I get this error when tryin both to launch it from Teleport dashboard and accessing the link directly -> "Access denied. Unable to serve application requests. Please try again. If the issue persists, verify if the Application Services are connected to Teleport."

Here is the teleport.yaml:
"
#
# A Sample Teleport configuration file.
#
# Things to update:
#  1. license.pem: You only need a license from https://dashboard.goteleport.com
#     if you are an Enterprise customer.
#
version: v3
teleport:
  nodename: teleport-vm
  data_dir: /var/lib/teleport
  log:
    output: stderr
    severity: INFO
    format:
      output: text
  ca_pin: ""
  diag_addr: ""
auth_service:
  enabled: "yes"
  listen_addr: 0.0.0.0:3025
  cluster_name: *******.com
  proxy_listener_mode: multiplex
ssh_service:
  enabled: "yes"
  commands:
  - name: hostname
    command: [hostname]
    period: 1m0s
proxy_service:
  enabled: "yes"
  web_listen_addr: 0.0.0.0:443
  public_addr: *******.com:443
  https_keypairs: []
  https_keypairs_reload_interval: 0s
  acme:
    enabled: "yes"
    email: info@*********.com
app_service:
    enabled: yes
    apps:
    - name: "nrtest"
      uri: "http://localhost:1880"
      public_addr: "nrtest.********.com"
"

 On the reinstall, I tried to add an app on the same server as the proxy/auth role. Not on another vm this time.
 Did run the tctl tokens add command and put the token in /tmp/token and ca-pin in /tmp/ca-pin. Got an error saying that I had to use proxy_server instead of auth_server:
"
teleportuser@teleport-vm:~$ sudo teleport app start \
   --token=/tmp/token \
   --ca-pin=tmp/ca-pin \
   --auth-server=*******.com:443 \
   --name=nrtest2 \
   --uri=http://nrtest2:1880
2023-02-14T17:29:02Z INFO             Starting Teleport v12.0.2 with a config file located at "/etc/teleport.yaml" common/teleport.go:517
ERROR: configuration error
initialization failed
config: when app_service is enabled, proxy_server must be specified instead of auth_server

teleportuser@teleport-vm:~$ sudo teleport app start \
   --token=/tmp/token \
   --ca-pin=tmp/ca-pin \
   --proxy_server=*****.com:443 \
   --name=nrtest2 \
   --uri=http://nrtest2:1880
teleport: error: unknown long flag '--proxy_server'

teleportuser@teleport-vm:~$ sudo teleport app start \
   --token=/tmp/token \
   --ca-pin=tmp/ca-pin \
   --proxy-server=******.com:443 \
   --name=nrtest2 \
   --uri=http://nrtest2:1880
teleport: error: unknown long flag '--proxy-server'

"
Did then procede to try adding app via Teleport dashboard / admin gui in browser:
"
 sudo teleport start --config=$HOME/.config/app_config.yaml
"
But get stuck with the same error as in the first post.

-frodes-bre

**Comment by zmb3** [comment]
This looks more like a support request that we can reproduce and troubleshoot, and it's hard to follow along without proper formatting on the configuration snippets. You might have more luck starting a GitHub discussion or joining our community Slack workspace.

A few things that stand out in your logs:

- _error: unknown long flag '--proxy-server'_: `teleport app start` only accepts an `--auth-server` flag. I recommend using a configuration file with `teleport start` instead.
- _connect: connection timed out_: have you verified that you can reach the proxy server on port 443 from your app service?

**Comment by frodes-bre** [comment]
Thanks! With regards to '--proxy-server', this was the error message I got when trying --auth-server flag:
"ERROR: configuration error
initialization failed
config: when app_service is enabled, proxy_server must be specified instead of auth_server". I will look into using a config file instead.

I also tried to add the app / app service on the same server that has the proxy server role on it.

-frodesbre

**Comment by Andre15711** [comment]
Any updates on this?
I have a simular problem and getting the error message
"Access denied. Unable to serve application requests. Please try again. If the issue persists, verify if the Application Services are connected to Teleport."

I installed teleport in a docker container. Using the latest docker image.

Log in to the teleport server itself via UI and ssh is not  problem at all. Only adding an app ist not working.

Any suggestions or ideas?

regards
André

**Comment by zmb3** [comment]
Closing due to inactivity
